desec.py

```diff
@@ -326,28 +327,54 @@ class APIClient:
 
     """
 
-    def __init__(self, token: str, retry_limit: int = 3):
+    def __init__(self, token, retry_limit=3, logger=getLogger("desec.client")):
```

Comment:
IMO `getLogger(__name__)` is a good name for a logger 95% of the time, because you are essentially recycling your (hopefully well-motivated) decision to put all code that uses this logger into this module.

The reasoning is: All this code is grouped together in this element of the code hierarchy (the module), so it probably makes sense that all this code emits logs to the equally-named element in the logger hierarchy. 

But that doesn't mean that there can't be a better name for a logger. Logger names can be a delicate decision, since they are more-or-less part of your public API. (I need to know what logger names emit what information so I can configure them adequately for my application).

In our situation explicitly, I would try to avoid just going with `getLogger(__name__)`, because we would be using the "root" of our namespace and that will make it harder to split things up should we want to. For my taste ~2k lines is already a tad to long for one file and I guess we will have more ideas down the line.

I don't see much of a problem with "optionally" passing loggers around. In this case I can also see a good usecase: If my application needs to use multiple different accounts, it can give differently named loggers to different client instances.

Reply:
We can take a hint about the intentional separation of logger name and module `__name__` from the fact that log records have an explicit field: `module` (see [here](https://docs.python.org/3/library/logging.html#logrecord-attributes)). So in my eyes `getLogger(__name__)` is really just a convention to fall back to an at least somewhat informative logger name whenever you encounter one of the only two hard problems in programming. ;)

Reply:
Oh, and keep/add type hints here ;)

Reply:
Done, `mypy desec.py` passes without any errrors now. :innocent: